subsys/net/l2/openthread/openthread.c

```diff
@@ -343,46 +343,50 @@ int openthread_start(struct openthread_context *ot_context)
 		otLinkSetPollPeriod(ot_context->instance, OT_POLL_PERIOD);
 	}
 
-	if (otDatasetIsCommissioned(ot_instance)) {
-		/* OpenThread already has dataset stored - skip the
-		 * configuration.
-		 */
-		NET_DBG("OpenThread already commissioned.");
-	} else if (IS_ENABLED(CONFIG_OPENTHREAD_JOINER_AUTOSTART)) {
-		/* No dataset - initiate network join procedure. */
-		NET_DBG("Starting OpenThread join procedure.");
-
-		error = otJoinerStart(ot_instance, OT_JOINER_PSKD, NULL,
-				      PACKAGE_NAME, OT_PLATFORM_INFO,
-				      PACKAGE_VERSION, NULL,
-				      &ot_joiner_start_handler, ot_context);
-
-		if (error != OT_ERROR_NONE) {
-			NET_ERR("Failed to start joiner [%d]", error);
-		}
+	if (!IS_ENABLED(CONFIG_OPENTHREAD_RCP)) {
```

Comment:
I'd rather exit early on `CONFIG_OPENTHREAD_RCP` instead of adding extra indentation level.

That also makes me wonder, how did it work with NCP, as all these configurations are supposed to be done by the Border Router and not the NCP itself. I think that a good option would be to add the following at the beginning of the function:
```
if (IS_ENABLED(CONFIG_OPENTHREAD_NCP) || IS_ENABLED(CONFIG_OPENTHREAD_RCP)) {
        return 0;
}
```

Going even further, I wonder if it actually makes sense to create a Zephyr network interface for the NCP/RCP application at all (since it's not supposed to use it directly), but this would need more investigation so let's leave it for now.

Reply:
If I exit early, I get linker errors : `undefined reference to `otThreadSetEnabled'. AFACT `IS_ENABLED(false)` is actually removed at compile-time.